src/Tribe/Log/Admin.php

```diff
@@ -0,0 +1,249 @@
+<?php
+class Tribe__Log__Admin {
+	public function __construct() {
+		add_action( 'wp_ajax_tribe_logging_controls', array( $this, 'listen' ) );
+		add_action( 'init', array( $this, 'serve_log_downloads' ) );
+	}
+
+	/**
+	 * Returns the HTML comprising the event log section for use in the
+	 * Events > Settings > Help screen.
+	 *
+	 * @return string
+	 */
+	public function display_log() {
+		$log_choices  = $this->get_available_logs();
+		$log_engines  = $this->get_log_engines();
+		$log_levels   = $this->get_logging_levels();
+		$log_entries  = $this->get_log_entries();
+		$download_url = $this->get_log_url();
+
+		$this->setup_script();
+
+		ob_start();
+		include trailingslashit( Tribe__Main::instance()->plugin_path ) . 'src/admin-views/event-log.php';
+		return ob_get_clean();
+	}
+
+	/**
+	 * Listens for changes to the event log settings updating and returning
+	 * an appropriate response.
+	 */
+	public function listen() {
+		$fields = wp_parse_args( $_POST, array(
+			'check'      => '',
+			'log-level'  => '',
+			'log-engine' => '',
+		) );
+
+		foreach ( $fields as &$single_field ) {
+			$single_field = filter_var( $single_field, FILTER_SANITIZE_STRING, FILTER_FLAG_ENCODE_HIGH );
+		}
+
+		if ( ! wp_verify_nonce( $fields['check'], 'logging-controls' ) ) {
+			return;
+		}
+
+		/**
+		 * Fires before log settings are committed.
+		 *
+		 * This will not happen unless a nonce check has already passed.
+		 */
+		do_action( 'tribe_common_update_log_settings' );
+
+		$this->update_logging_level( $fields['log-level'] );
+		$this->update_logging_engine( $fields['log-engine'] );
+
+		/**
+		 * Fires immediately after log settings have been committed.
+		 */
+		do_action( 'tribe_common_updated_log_settings' );
+
+		$data = array(
+			'logs' => $this->get_available_logs(),
+		);
+
+		if ( ! empty( $fields['log-view'] ) ) {
+			$data['entries'] = $this->get_log_entries( $fields['log-view'] );
+		}
+
+		wp_send_json_success( $data );
+	}
+
+	/**
+	 * Sets the current logging level to the provided level (if it is a valid
+	 * level, else will set the level to 'default').
+	 *
+	 * @param string $level
+	 */
+	protected function update_logging_level( $level ) {
+		$this->log_manager()->set_level( $level );
+	}
+
+	/**
+	 * Sets the current logging engine to the provided class (if it is a valid
+	 * and currently available logging class, else will set this to null - ie
+	 * no logging).
+	 *
+	 * @param string $engine
+	 */
+	protected function update_logging_engine( $engine ) {
+		try {
+			$this->log_manager()->set_current_logger( $engine );
+		}
+		catch ( Exception $e ) {
+			// The class name did not relate to a valid logging engine
+		}
+	}
+
+	/**
+	 * Adds a script to handle the event log settings.
+	 */
+	protected function setup_script() {
+		wp_enqueue_script(
```

Comment:
That was something that me and @borkweb were talking about past few days as something that would be a easy win for us in terms of having a quick way to enqueue scripts only when they are needed but registering everything on all pages.